### PR TITLE
Xthin improvements

### PR DESCRIPTION
**Pull request by zander**
Following a review of xthinblocks I introduce various cleanups and changes.

### CODE REVIEW

**Comment by zander** [comment]
It would be wonderful if guys like  Peter Tschipper(@ptschip), Andrew Stone(@gandrewstone), or @rusty-loy could give a little attempt at proof reading this.  Maybe steal some of this, if its good, and perhaps suggest more fixes based on work that may have happened elsewhere...

Thanks!

**Comment by zander** [comment]
I got some reviews and comments on slack, these two certainly are interesting to follow up on (thanks dagurval for them!)
- I don’t agree that xthin requests require DoS limitations, when merkleblock requests don’t. 
- I think connect-thinblocks and all code related should be removed. There are more than 200 xthin supporting nodes on the network already, regular addnode probably works fine.

File: src/bloom.h [hunk 1]
```diff
@@ -70,7 +70,7 @@ class CBloomFilter
     CBloomFilter(unsigned int nElements, double nFPRate, unsigned int nTweak, unsigned char nFlagsIn);
     CBloomFilter() : isFull(true), isEmpty(false), nHashFuncs(0), nTweak(0), nFlags(0) {}
 
-    ADD_SERIALIZE_METHODS;
+    ADD_SERIALIZE_METHODS
```
Comment (rebroad) [inline src/bloom.h:73]: why remove the semicolon?

**Comment by rebroad** [comment]
Which are the previous XThin pull requests please?

**Comment by rebroad** [comment]
@zander What vulnerability in Core's p2p messaging system please? (re bloom)

File: src/main.cpp [hunk 2]
```diff
@@ -6021,6 +6021,10 @@ bool SendMessages(CNode* pto)
         if (pindexBestHeader == NULL)
             pindexBestHeader = chainActive.Tip();
         bool fFetch = state.fPreferredDownload || (nPreferredDownload == 0 && !pto->fClient && !pto->fOneShot); // Download if this is a nice peer, or we have no nice peers and this one might do.
+
+        if (fFetch)  // Don't use peer unless our chain is shorter
```
Comment (rebroad) [inline src/main.cpp:5937]: Why isn't this in Core?!

**Comment by zander** [comment]
> What vulnerability in Core's p2p messaging system please? (re bloom)

Ehm, I have no idea what you are asking about. Where did anyone talk about vulnerabilities?

Also, Reviewing old code and commits that have been long since superseded isn't helpful.

File: src/main.cpp (hunk 2 above)
Comment (dexX7) [reply src/main.cpp:5937]: Looks like this was reverted via afc70ad877d38dae376e03a79f6743dda3b2128c.

**Pull request merged**